### ISSUE TITLE
train_batch_size

### ISSUE THREAD

**Issue by mandyyyyii** [OpenRLHF/OpenRLHF#397]
What is the difference between train_batch_size and micro_train_batch_size?

If I set micro_train_batch_size as 2 and use 8 gpu, the final batch size will be 16 and it seems that train_batch_size doesn't play a role in the code.

**Comment by hijkzzz** [comment]
train_batch_size = micro_train_batch_size  * gpus * accumulated_gradient

**Comment by iMurphL** [comment]
In my PPO training, I couldn't find where the accumulated_gradient takes effect. Is it that PPO training does not support gradient accumulation?

**Comment by fan-yang1** [comment]
> In my PPO training, I couldn't find where the accumulated_gradient takes effect. Is it that PPO training does not support gradient accumulation?

Gradient accumulation is managed by DeepSpeed. We only need to pass the argument to DeepSpeed. This is handled in openrlhf/utils/deepspeed/deepspeed.py within the get_ds_train_config function. The ds_config is then passed to deepspeed.initialize in _ds_init_train_model function. see also https://www.deepspeed.ai/docs/config-json/
